libs/langchain/langchain/utilities/cube.py

```diff
@@ -0,0 +1,338 @@
+                information += "| --- | --- | --- | --- |\n"
+
+                for measure in m["measures"]:
+                    information += (
+                        f"| {measure.get('shortTitle')} "
+                        f"| {measure.get('description', '')} "
+                        f"| {measure.get('name')} "
+                        f"| {measure.get('type')} |\n"
+                    )
+
+            if m["dimensions"]:
+                information += "\n### Dimensions:\n"
+                information += "| Title | Description | Column | Type |\n"
+                information += "| --- | --- | --- | --- |\n"
+
+                for dimension in m["dimensions"]:
+                    information += (
+                        f"| {dimension.get('shortTitle')} "
+                        f"| {dimension.get('description', '')} "
+                        f"| {dimension.get('name')} "
+                        f"| {dimension.get('type')} |\n"
+                    )
+
+            models.append(information)
+
+        models.sort()
+
+        return "\n\n".join(models)
+
+    def load(self, query: Union[Dict[str, Any], str, Query]) -> Dict[str, Any]:
+        """Load of the cube.
+        See https://cube.dev/docs/reference/rest-api#v1load for more information.
+        """
+
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": self.cube_api_token.get_secret_value(),
+        }
+
+        params = {"query": ""}
+
+        if isinstance(query, Query):
+            params["query"] = query.json(exclude_unset=True)
+        elif isinstance(query, Dict):
+            params["query"] = json.dumps(query)
+
+        response = requests.post(
+            url=f"{self.cube_api_url}/load",
+            headers=headers,
+            json=params,
+        )
+
+        raw = response.json()
+        if "error" in raw:
+            raise ValueError(raw["error"])
+        else:
+            response.raise_for_status()
+
+        return raw
+
+    def sql(self, query: Union[Dict[str, Any], Query]) -> Dict[str, Any]:
+        """SQL of the cube.
+        See https://cube.dev/docs/reference/rest-api#v1sql for more information.
+        """
+
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": self.cube_api_token.get_secret_value(),
+        }
+
+        params = {"query": ""}
+
+        if isinstance(query, Query):
+            params["query"] = query.json(exclude_unset=True)
+        elif isinstance(query, Dict):
+            params["query"] = json.dumps(query)
+
+        response = requests.get(
+            url=f"{self.cube_api_url}/sql",
+            headers=headers,
+            params=params,
+        )
+
+        raw = response.json()
+        if "error" in raw:
+            raise ValueError(raw["error"])
+        else:
+            response.raise_for_status()
+
+        return raw["sql"]
```

Comment:
Could you please explain why do you need the `/sql` endpoint?

Reply:
Sometimes, we do not wish to directly query data; we simply want to obtain the SQL. Subsequently, there might be a need to make secondary modifications to the SQL based on certain business requirements. That's why we are adding a /sql endpoint.